src/vaults/VaultManager.ts

```diff
@@ -340,7 +350,7 @@ class VaultManager {
     );
     const vaultIdString = vaultId.toString() as VaultIdString;
     return await this.vaultLocks.withF(
-      [vaultId.toString(), RWLockWriter, 'write'],
+      [vaultId.toString(), RWLockWriter, 'write', ctx],
```

Comment:
I've done this in a couple of places, but I don't think this is the right way to do this. The type itself is also pretty complex and I can't understand it. This feels wrong to me, but no error is raised here.